pkg/kubelet/status/status_manager.go

```diff
@@ -889,6 +916,16 @@ func mergePodStatus(oldPodStatus, newPodStatus v1.PodStatus, couldHaveRunningCon
 			newPodStatus.Phase = oldPodStatus.Phase
 			newPodStatus.Reason = oldPodStatus.Reason
 			newPodStatus.Message = oldPodStatus.Message
+			if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+				// revert setting of the pod disruption condition until the pod is terminal in order to do not issue
+				// an unnecessary PATCH request
+				revertPodCondition(&oldPodStatus, &newPodStatus, v1.AlphaNoCompatGuaranteeDisruptionTarget)
```

Comment:
Similar as above comment, this revert logic is a bit ugly... we already reinitialize the pod conditions on line 871. Instead of this revert logic can we just guard the adding of the condition in the first place? (see earlier comment above)

Reply:
Indeed, however this way we avoid duplication of the logic which checks there is transition into Failed phase and the couldHaveRunningContainers. And this seems more consistent with the current approach to revert the phase, reason and message fields.

Reply:
As above, agree the reverting is ugly and applied the suggestion, after some modifications. In particular, extracted the `transitioningToTerminalPhase` bool for reuse.